### PR TITLE
:sparkles: add products

### PR DESCRIPTION
**Pull request by sebastianMindee**
### Changes
* :sparkles: add support for all missing products
* :arrow_up: manually add tests for FinDoc
* :arrow_up: fix & add tests for cropper extra
* :arrow_up: update test lib
* :recycle: misc fixes
* :arrow_up: add more sturdy tests
* :memo: add full documentation

### CODE REVIEW

File: src/Product/Receipt/ReceiptV5Document.php [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+:Total Amount: $this->totalAmount
+:Total Net: $this->totalNet
+:Total Tax: $this->totalTax
+:Tip and Gratuity: $this->tip
+:Taxes: $this->taxes
+:Supplier Name: $this->supplierName
+:Supplier Company Registrations: $supplierCompanyRegistrations
+:Supplier Address: $this->supplierAddress
+:Supplier Phone Number: $this->supplierPhoneNumber
+:Line Items: $lineItemsSummary
```
Comment (ianardee) [inline src/Product/Receipt/ReceiptV5Document.php:182]: lineItems should be stringable

```suggestion
:Line Items: $this->lineItems
```

File: src/Product/Receipt/ReceiptV5LineItem.php [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+namespace Mindee\Product\Receipt;
+
+use Mindee\Parsing\Common\SummaryHelper;
+use Mindee\Parsing\Standard\FieldConfidenceMixin;
+use Mindee\Parsing\Standard\FieldPositionMixin;
+
+/**
+ * List of line item details.
+ */
+class ReceiptV5LineItem
```
Comment (ianardee) [inline src/Product/Receipt/ReceiptV5LineItem.php:12]: should explicitly implement the `Stringable` interface

> As of PHP 8.0.0, any class that contains a [__toString()](https://www.php.net/manual/en/language.oop5.magic.php#object.tostring) method will also implicitly implement the [Stringable](https://www.php.net/manual/en/class.stringable.php) interface, and will thus pass type checks for that interface. Explicitly implementing the interface anyway is recommended.

https://www.php.net/manual/en/language.oop5.magic.php#object.tostring

```suggestion
class ReceiptV5LineItem implements Stringable
```

File: src/Product/Receipt/ReceiptV5LineItems.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Mindee\Product\Receipt;
+
+/**
+ * List of line item details.
+ */
+class ReceiptV5LineItems extends \ArrayObject
```
Comment (ianardee) [inline src/Product/Receipt/ReceiptV5LineItems.php:8]: should implement `Stringable`

File: src/Product/Receipt/ReceiptV5LineItems.php [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        return $outStr . "+";
+    }
+
+
+    /**
+     * String representation.
+     *
+     * @return string
+     */
+    public function lineItemsToStr(): string
```
Comment (ianardee) [inline src/Product/Receipt/ReceiptV5LineItems.php:45]:
```suggestion
    public function __toString(): string
```

**Review by ianardee — CHANGES_REQUESTED** [review]

File: src/Product/Receipt/ReceiptV5LineItem.php (hunk 2 above)
Comment (sebastianMindee) [reply src/Product/Receipt/ReceiptV5LineItem.php:12]: Doesn't exist in 7.4 :upside_down_face:

File: src/Product/Receipt/ReceiptV5LineItems.php (hunk 4 above)
Comment (sebastianMindee) [reply src/Product/Receipt/ReceiptV5LineItems.php:45]: Oh yeah that makes more sense :sweat_smile:

**Review by ianardee — APPROVED** [review]

**Pull request merged**